### ISSUE TITLE
Cryptic error messages without build-essential

### ISSUE THREAD

**Issue by paemurru** [oscar-system/oscar-website#315]
When the user installs OSCAR on Ubuntu or Debian and forgets to install `build-essential`, they will have strange error messages. Perhaps it would be better to check whether `build-essential` is installed (meaning, whether specific C files exist that OSCAR uses) and if not, give instructions to install it.

**Comment by fingolfin** [comment]
What are those strange error messages?

In how far is this an issue with the website, isn't this more a feature request for Oscar.jl itself? (I am not sure we'd do it, though; that would require adding details about specific distros to Oscar.jl. But depending on what the "strange errors" are, there might be other things we could check for and suggest to the user as workaround or solution.

**Comment by HereAround** [comment]
We could add a sentence to the installation instructions for Ubuntu/Debian like: "Please install build-essential, see here(*fill in link*) for instructions." Would this close this issue @paemurru ?

**Comment by benlorenz** [comment]
> We could add a sentence to the installation instructions for Ubuntu/Debian like: "Please install build-essential, see here(_fill in link_) for instructions." Would this close this issue @paemurru ?

We do have this:
![install](https://github.com/oscar-system/oscar-website/assets/1437056/b1343d94-d5f4-46e6-bb77-85ef6d29113f)

**Comment by paemurru** [comment]
@simonbrandhorst commented to me that a typical user would install a Julia package by `add Oscar`. If they are using a Debian-based system and `build-essential` is missing, then OSCAR will not run correctly and will output strange error messages. If the most common way of installing a package fails for OSCAR, I see that as a problem.

**Comment by simonbrandhorst** [comment]
Same here. Nobody reads manuals unless they really have to. Imagine having to read a manual for each julia package you want to install. 

Just `add Oscar` and `using Oscar` should work for 99% of users no fuss. 
If `build-essential` cant be avoided, the error message should say how to get it/point to the place in the manual.

**Comment by HereAround** [comment]
To clarify: Am I right to assume that you do *not* want changes to the website? (All that I could see in this regard we already have, as @benlorenz pointed out above.)

**Comment by simonbrandhorst** [comment]
Yes, the website is perfectly fine and well written. Just the error message should be more helpful. (or ideally just build-essentials installed automatically? But I have no clue about the  underlying technical considerations)

**Comment by benlorenz** [comment]
Seeing such an error message (as requested before) would be really helpful to understand what we need to check and where to add a corresponding message.
And no this cannot be installed automatically.

**Comment by paemurru** [comment]
@benlorenz @fingolfin These error messages appeared while installing on a laptop that I do not have access to anymore. If you wish to see the specific error messages, please install OSCAR on a Debian-based system without `build-essential`. For example, on Debian Live in a virtual machine or using a bootable Debian Live from a flash drive.

**Comment by paemurru** [comment]
@HereAround The ideal solution would be if OSCAR provided helpful error messages that instruct to install `build-essential`. If this is done, then no change for the website would be necessary.

If this is for some reason infeasible, then perhaps it would help if there is a "common installation problems" section (or something like that) on the website

**Comment by fingolfin** [comment]
There are *hundreds* of Linux distributions. Debian is common on servers but unusual for regular users.

I don't think it would be a good use of the scarce time of @benlorenz or other of our devs to sit down, install a Debian and try to reproduce an error just to deal with this border case. And "deal" would probably boil down to adding a hard coded lists of checks for specific distros (if we can even reliably detect them), calling out to package manager tools to query the presence of certain packages, etc. -- that's not a road I'd wish for us to go down to.

Moreover the issue might even be a resolve one -- e.g. at some point we compiled GAP packages during installation of OSCAR (we still might do). This is something that will soon go away, too. Perhaps there are more issues left -- then I hope the next person to report them will from the start including the error messages (or provide them immediately when requested) and we can look into resolving it then.

Closing this for now
